### PR TITLE
[TS] LPS-181899 Item Selector won't show all Sites grouped to the paginated page

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

I have updated the solution as per Brian's [suggestion](https://github.com/brianchandotcom/liferay-portal/pull/134317#issuecomment-1534795525).

Please note that I have extracted the permission check into its own method, `_hasViewPermission()` because the `PortalException` thrown by `GroupPermissionUtil.contains()` cannot be handled in the lambda.

Best regards,
István

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+				group.isCompany() ||
+				permissionChecker.isGroupAdmin(group.getGroupId()) ||
+				GroupPermissionUtil.contains(
+					permissionChecker, group, ActionKeys.VIEW);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
+		}
+
+		return hasViewPermission;
```
Comment (adolfopa) [inline modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java:133]: Can you avoid modifying the variable? Something like:
```
try {
    return group.isCompany() || ...
}
catch (...) {
    ...
    return false;
}
```

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java (hunk 1 above)
Comment (IstvanD) [reply modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java]: I have changed it according to your suggestion. Then the source formatter rewrote it to its current form.
See the pushed code.

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**